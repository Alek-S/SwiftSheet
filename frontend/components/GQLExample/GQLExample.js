import React from 'react';
import PropTypes from 'prop-types';
import { ApiCard } from '../../defaultStyle';

/**
 * @function
 * Card providing link to GraphQL api endpoint & body of sheet
 * @param {string} - props.host - host origin
 * @param {string} - props.id - sheet ID
 * @returns {jsx} <RestExample />
 */
const GQLExample = ({ host, id }) => (
	<ApiCard>
		<p>
			<span className="pill">GraphQL API:</span>
			<a href={`${host}/graphql`} target="_blank">
				{host}/graphql
			</a>
		</p>
		<p className="query_title">
			{' '}
			<span className="pill-secondary">Body:</span>
			<span className="gql_query">
				{'{'}
				sheet( _id: "{id}" ){'{'}
				sheetData
				{'}'}
				{'}'}
			</span>
		</p>
	</ApiCard>
);

GQLExample.propTypes = {
	host: PropTypes.string.isRequired,
	id: PropTypes.string.isRequired,
};

export { GQLExample };
