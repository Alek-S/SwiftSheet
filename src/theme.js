/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
