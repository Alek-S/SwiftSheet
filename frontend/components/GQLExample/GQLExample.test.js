import React from 'react';
import 'jest-styled-components';

import { GQLExample } from './GQLExample';
import theme from '../../theme';

describe('GQLExample', () => {
	test('matches the snapshot', () => {
		const tree = shallowWithTheme(
			<GQLExample host="mockHost.com" id="42" />,
			theme
		).dive();
		expect(toJson(tree)).toMatchSnapshot();
	});
});
