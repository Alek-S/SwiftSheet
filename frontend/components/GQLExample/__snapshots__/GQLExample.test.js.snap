// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GQLExample matches the snapshot 1`] = `
<defaultStyle__ApiCard>
  <p>
    <span
      className="pill"
    >
      GraphQL API:
    </span>
    <a
      href="mockHost.com/graphql"
      target="_blank"
    >
      mockHost.com
      /graphql
    </a>
  </p>
  <p
    className="query_title"
  >
     
    <span
      className="pill-secondary"
    >
      Body:
    </span>
    <span
      className="gql_query"
    >
      {
      sheet( _id: "
      42
      " )
      {
      sheetData
      }
      }
    </span>
  </p>
</defaultStyle__ApiCard>
`;
